lib/components/Web3Context/types.ts

```diff
@@ -71,6 +72,9 @@ export type Web3State = ConnectedWeb3State | DisconnectedWeb3State;
 export type Web3ModalState = Web3State & {
   connect?: (wallet?: string) => Promise<void>;
   disconnect?: () => Promise<void>;
+  showModal?: boolean;
+  renderModal?: FC;
+  toggleModal?: () => void;
```

Comment:
set more strict types so you don't have to keep checking for existence everywhere you use it
```suggestion
  showModal: boolean;
  renderModal: () => ReactNode;
  toggleModal: () => void;
```

And then on L80 web3InitialState provide dummy defaults
```
{
  showModal: false,
  renderModal: () => null,
  toggleModal: () => {},
}
```

Reply:
updated